### PR TITLE
Operational mission

### LINKED ISSUE #223: FAST-OAD notebooks: update input XML file for working with mission module
The `CeRAS01_baseline.xml` file provided with notebooks does not contain inputs that are needed by the mission performance module.

Then, when activating the mission module instead of the Breguet module, the input generation produces some NaN values in the XML file. Using this incomplete input file leads to following error:

```python
...

~/.pyenv/versions/3.7.7/envs/fastoad-test/lib/python3.7/site-packages/fastoad/utils/physics/atmosphere.py in __init__(self, altitude, delta_t, altitude_in_feet)
     76         # For convenience, let's have altitude as numpy arrays and in meters in all cases
     77         unit_coeff = foot if altitude_in_feet else 1.0
---> 78         self._altitude = np.asarray(altitude) * unit_coeff
     79 
     80         # Sets indices for tropopause

TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'

```

### PR DESCRIPTION
**Pull request by christophe-david**
This PR proposes an extension of the YAML-defined mission file for computing missions inside the sizing loop (design mission) and outside the sizing loop (operational missions).
It also unifies the performance computation, by allowing to specify the legacy Breguet algorithm through a YAML mission file.

It also fixes #223.

A documentation for the mission module has also been written:
https://fast-oad.readthedocs.io/en/operational-mission/documentation/mission_module/index.html
@Martin-Delavenne , I asked you as reviewer especially so you can provide feedback on the documentation.

Not addressed in this PR:
- to allow a complete mission specification, the process will need a takeoff segment and the specification of a starting point. Currently, the documentation does not mention these gaps, as I hope to fill them before the 1.0 release.
- A JSON schema validation would be useful to ease file editing. Doing will also question the current validation with `strictYAML` (though strictYAML allows some checks the JSON schema won't)

___

**Additional information:**
A convergence problem on the optimization case showed that the default tolerance (`atol=1e-10`) for NLBGS termination is far too demanding. Setting it to 1e-8 is enough to get back the convergence, and setting it to 1e-2 is enough to keep consistent results (discrepancies are less than 1 gram on MTOW). And of course, relaxing tolerance allows a (slightly) lower CPU consumption.

### CODE REVIEW

**Comment by Martin-Delavenne** [comment]
Few very general comments about the doc: 

- In the Mission file paragraph, routes section you mention that routes are identified by 4 arguments : "climb_parts", "cruise_part", "descent_parts" and "range". But in the example, when you are describing the diversion_route the argument "parts" appears. If the argument "parts" works with routes keyword shouldn't it be added to the list of admitted arguments? If a route can be alternatively described through the "parts" arguments containing different phases and a cruise segment I think it should also be added in the doc for a greater clarity. 
- In the same example as cited above, the cruise segment is once called with the "cruise" argument the second time with "cruise_type" but this argument is never described later in the doc. 
- In the Segment paragraph, speed_change section, no example is provided.

**Comment by christophe-david** [comment]
Thank you @Martin-Delavenne for the comments. My replies below:

> Few very general comments about the doc:
> 
> * In the Mission file paragraph, routes section you mention that routes are identified by 4 arguments : "climb_parts", "cruise_part", "descent_parts" and "range". But in the example, when you are describing the diversion_route the argument "parts" appears. If the argument "parts" works with routes keyword shouldn't it be added to the list of admitted arguments? If a route can be alternatively described through the "parts" arguments containing different phases and a cruise segment I think it should also be added in the doc for a greater clarity.
> * In the same example as cited above, the cruise segment is once called with the "cruise" argument the second time with "cruise_type" but this argument is never described later in the doc.

Actually, the example is wrong. Using `parts` keyword in route definition was my first thought, but since routes are expected to have this particular structure, I considered the definition schema should show it more clearly. Same for `cruise_type`, it was part of the previous syntax and is no more relevant. I will fix the example.

> * In the Segment paragraph, speed_change section, no example is provided.

Indeed. I'll fix that also.

**Review by ScottDelbecq — COMMENTED** [review]
Thanks for this great job! Here are my first comments on the docs, my feedback on the code will come soon!

File: docs/documentation/mission_module/mission_file/segments.rst [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+:code:`cruise`
+==============
+
+A :code:`cruise`segment simulates a flight part at constant speed and altitude, and regulated
+thrust rate (drag is compensated).
+
+Optionally, target altitude can be set to :code:`optimal_flight_level`. In such case, cruise will
+be preceded by a climb segment that will put the aircraft at the altitude that will minimize the
+fuel consumption for the whole segment (including the prepending climb).
+This option is available because the :ref:`segment-altitude_change`segment can reach an altitude
```
[the remark this answers is not in the document]
Comment (christophe-david) [reply docs/documentation/mission_module/mission_file/segments.rst]: Merci. corrigé en 7cb7eaf013dbae5deeb20a4a571b7e97b1c2c3b1.

File: docs/documentation/mission_module/mission_file/segments.rst [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+      # altitude: constant                      # Not needed, because assumed by default
+      time:
+        value: 20                               # 20 minutes holding
+        unit: min
+
+
+:code:`taxi`
+============
+
+A :code:`taxi`segment simulates the mission parts between gate and takeoff or landing, at constant
```
[the remark this answers is not in the document]
Comment (christophe-david) [reply docs/documentation/mission_module/mission_file/segments.rst]: Merci. corrigé en 7cb7eaf013dbae5deeb20a4a571b7e97b1c2c3b1.

File: docs/documentation/mission_module/mission_module.rst [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+ - :code:`data:mission:<mission_name>:needed_block_fuel`: Burned fuel during mission. Output.
+ - :code:`data:mission:<mission_name>:block_fuel`: Actual block fuel. Input or output, depending on options below.
+
+
+
+************************************
+Usage in FAST-OAD configuration file
+************************************
+
+The mission module can be used with the identifier :code`"fastoad.performances.mission"`.
```
[the remark this answers is not in the document]
Comment (christophe-david) [reply docs/documentation/mission_module/mission_module.rst]: Et encore merci. corrigé en 7cb7eaf013dbae5deeb20a4a571b7e97b1c2c3b1 :)

File: docs/documentation/mission_module/mission_file/segments.rst [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+
+
+
+.. _segment-parameter-engine_setting:
+
+:code:`engine_setting`
+======================
+
+Expected value for :code:`engine_setting` are :code:`takeoff`, :code:`climb`
+, :code:`cruise` or :code:`idle`
```
[the remark this answers is not in the document]
Comment (christophe-david) [reply docs/documentation/mission_module/mission_file/segments.rst:274]: J'ai rajouté un brin d'explication en 0775dc164b8f22e85fd744a66769b17746173358. Dis-moi si ça te semble suffisant.

Mais au fond, c'est sans doute le rubber engine qu'il faudra mieux documenter.

[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply docs/documentation/mission_module/mission_file/segments.rst:274]: Oui parfait mettre le lien ver rubber engine est très utile.

File: tests/integration_tests/oad_process/test_oad_process.py [hunk 5]
```diff
@@ -13,7 +13,6 @@
 #  GNU General Public License for more details.
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <https://www.gnu.org/licenses/>.
-import os
```
Comment (ScottDelbecq) [inline tests/integration_tests/oad_process/test_oad_process.py:16]: `os` is used on line 62.

**Review by ScottDelbecq — CHANGES_REQUESTED** [review]
Thanks for this major work! 
This is a rapid feedback on the usage.
Everything seems to work properly with acceptable computation times except regarding my single comment.

However, I noticed that on the last optimizer iteration in `test_api_optim()` and in the optimization part of the tutorial notebook the MDA fails to converge (even with 300 max. iterations). Since we test the mass breakdown consistency, I guess it is the wing area which is not consistent?

This is not the case when we use the mission file:

```toml
    [model.performance]
        id = "fastoad.performances.mission"
        propulsion_id = "fastoad.wrapper.propulsion.rubber_engine"
        # mission_file_path = "::sizing_breguet"
        mission_file_path = "::sizing_mission"
        out_file = "./flight_points.csv"
        adjust_fuel = true
        is_sizing = true
```

File: tests/integration_tests/oad_process/test_oad_process.py (hunk 5 above)
Comment (christophe-david) [reply tests/integration_tests/oad_process/test_oad_process.py:16]: Fixed in 416455c97505585c058954f152147b57a9168d58

**Comment by christophe-david** [comment]
> However, I noticed that on the last optimizer iteration in `test_api_optim()` and in the optimization part of the tutorial notebook the MDA fails to converge (even with 300 max. iterations). Since we test the mass breakdown consistency, I guess it is the wing area which is not consistent?

Thanks for noticing that. I fixed it and updated PR description about that subject.

**Pull request merged**